blueprints/addon/index.js

```diff
@@ -112,13 +112,16 @@ module.exports = {
     this.ui.writeLine(prependEmoji('✨', `Creating a new Ember addon in ${chalk.yellow(process.cwd())}:`));
   },
 
-  afterInstall() {
+  afterInstall(options) {
     let packagePath = path.join(this.path, 'files', 'package.json');
     let bowerPath = path.join(this.path, 'files', 'bower.json');
 
     [packagePath, bowerPath].forEach((filePath) => {
       fs.removeSync(filePath);
     });
+
+    let pathToRemove = options.ciProvider !== 'travis' ? '.travis.yml' : '.github';
+    fs.removeSync(`${this.project.root}/${pathToRemove}`);
```

Comment:
Some updates here!  I've overriden the files method for app and addon blueprints.  One change I needed to make was to pass options through the files function b/c the this context and whether or not this.options.ciProvider was available (options is set in models/blueprint) wasn't consistent - e.g. can be called from blueprints/addon/index.

The other option I dug into was filesToRemove instead of files.  Ideally this is a getter instead of a property and I can access stateful information about the options.ciProvider and adjust accordingly.  However, the this context is not correct - e.g. blueprints/app/index and not lib/models/blueprint.  Would changing filesToRemove to a function and passing in options be a breaking change?

Overall on both points, the lack of statefulness in these commands caused a bit larger changes than necessary.  But perhaps that is by design - to compose functions with the arguments as state.